docs/cookbook/links.rst

```diff
@@ -0,0 +1,221 @@
+            }
+        }
+    }
+
+Here is a another example of using forward link. This time we only return
+last names of the artists as plain string (not an object). In this case, we
+need to alias the field with ``:=``:
+
+.. code-block:: edgeql-repl
+
+    tutorial> SELECT Movie {
+    .........     title,
+    .........     starring := Movie.actors.last_name,
+    ......... };
+    {Object {
+        title: 'Blade Runner 2049',
+        starring: {
+            'Ford',
+            'Gosling',
+            'de Armas',
+        }
+    }}
+
+To find all movies that a person is starred in we use a **backward link**
+traversal ``.<`` operator:
+
+.. code-block:: edgeql-repl
+
+    tutorial> SELECT Person {
+    .........     first_name,
+    .........     movies := Person.<actors[IS Movie].title,
+    ......... } FILTER .first_name = 'Ryan';
+    {Object {
+        first_name: 'Ryan',
+        movies: {
+            'Blade Runner 2049',
+        }
+    }}
+
+You might also note that we've added ``[IS Movie]`` type filter. This is how
+backward link traversal works: EdgeDB fetches every object in the entire
+database having the field ``actors`` which is a ``Person``. So we narrow down
+the set of objects to ``Movie`` and select a title from it.
+
+All other tools work on backward link:
+
+.. code-block:: edgeql-repl
+
+    tutorial> SELECT Person {
+    .........     first_name,
+    .........     movies := Person.<actors[IS Movie] { title, year }
+    ......... } FILTER .first_name = 'Ryan';
+    {Object {
+        first_name: 'Ryan',
+        movies: {
+            Object { title: 'Blade Runner 2049', year: 2017 },
+        }
+    }}
+
+Or more complex example:
+
+.. code-block:: edgeql-repl
+
+    tutorial>     SELECT Person {
+    .........         first_name,
+    .........         colleagues := Person.<actors[IS Movie].actors {
+    .........             first_name
+    .........         }
+    .........     } FILTER .first_name = 'Ryan';
+    {
+        Object {
+            first_name: 'Ryan',
+            colleagues: {
+                Object { first_name: 'Ana' },
+                Object { first_name: 'Harrison' },
+                Object { first_name: 'Ryan' },
+            }
+        }
+    }
+
+Now you may notice that the Ryan (Gosling) is mentioned as a colleague of
+himself. To fix it we need few more concepts.
+
+First note that the request above is an equivalent of:
+
+.. code-block:: edgeql-repl
+
+    tutorial> SELECT Person {
+    .........     first_name,
+    .........     colleagues := (
+    .........         SELECT Person.<actors[IS Movie].actors {
+    .........             first_name,
+    .........         }
+    .........     ),
+    ......... } FILTER .first_name = 'Ryan';
+
+Note: we wrapped a backward link access by ``SELECT`` subquery.
+
+Still we can't filter out by ``Person != Person`` because EdgeDB can't
+distinguish them. To make that work we should give the inner query an alias:
```

Comment:
@elprans Is this the way you'd phrase this? Specifically "``Person != Person`` because EdgeDB can't
distinguish them" doesn't sound correct to me.

Reply:
This paragraph is actually outdated.  The query has been reformulated.

Reply:
".. distinguish the object on the left and the object on the right." ? 

This is a bit ugly wording I agree, but other than that it's not outdated.

Reply:
But it is. You no longer filter `Person != Person`, you do `Peer != Person`, and that is not ambiguous.

Reply:
The idea of this introduction is to show "why" you need to create an alias. Because you could add a FILTER right away if only you have a way to distinguish a Person from the inner query and a Person from the outer query.

Reply:
You don't have to create an alias.  You're still thinking of a previous formulation that was incorrect.

This is sufficient:

```
    tutorial> SELECT Person {
    .........     first_name,
    .........     colleagues := (
    .........         SELECT Person.<actors[IS Movie].actors {
    .........             first_name,
    .........         } FILTER Person.<actors[IS Movie].actors != Person
    .........     ),
    ......... } FILTER .first_name = 'Ryan';
```

Aliasing simply makes the query more readable (which is, btw, one of big reasons why aliases exist and why they work the way they work).

Reply:
Well, I'm not thinking of a previous formulation. I was just unaware that it's possible :) (there are cases in SQL where the alias is required, perhaps, thats why)